src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1142,5 +1142,21 @@ internal static bool IsAzureSqlServerEndpoint(string dataSource)
 
             return false;
         }
+
+        static internal ArgumentOutOfRangeException InvalidDataRowVersion(DataRowVersion value)
+        {
+#if DEBUG
```

Comment:
@stephentoub guessing @gafter will have a more detailed reason.  

My intuition though is that this just isn't a common enough case to think about.  In a normal switch it's common to have empty branches (simple break case).  In the presence of non-empty switches you still need to generate the branch for the empty one if only to stop it from going to one that has code.  

This case is when all branches are empty.  Probably not common enough to optimize.